docs/definingworkflows.rst

```diff
@@ -365,7 +393,59 @@ To ease composability and avoid unwanted collisions, each Stage is defined withi
 Composition using Subworkflows
 ------------------------------
 
+Subworkflows facilitates the definition of an entire workflow to run as a single step inside another workflow. Nesting workflows are very useful, especially when sections of code are run in parallel and are executed independently from each other. 
+
+One example is a workflow that defines two unique steps, the second depending on the first. The second step expects a file from the first step as input, and it will wait until everything in step one is finished before starting. The first step is called as a multistep-stage and so a number of instances are scheduled of the same packtivity. Without a subworfklow the second step would have to wait until all instances are done in step one before proceeding, despite only depending on the output from one of them. 
+
+Fortunately with a subworkflow, steps with scattered input can proceed independently. Instances that have finished step one can continue to step two of the workflow, even though other instances are still running step one.
+
+When defining a subworkflow to run as a single step inside another workflow, use the property :code:`workflow`.
+
+Example: ::
+
+  # workflow.yml
+  stages:
+    - name: map
+      dependencies: [init]
+      scheduler:
+        scheduler_type: multistep-stage
+        parameters:
+          input: {step: init, output: input, unwrap: true}
+        workflow: {$ref: subflow.yml}
+        scatter:
+          method: zip
+          parameters: [input]
+      
+  # subflow.yml    
+  stages:
+    - name: stageA
+      dependencies: ['init']
+      scheduler:
+        scheduler_type: singlestep-stage
+        parameters:
+          input_file: {step: init, output: input}
+          output_file: '{workdir}/output'
+        step: {$ref: steps.yml#/stepA}
+    - name: stageB
+      dependencies: ['stageA']
+      scheduler:
+        scheduler_type: singlestep-stage
+        parameters:
+          input_file: {stages: stageA, output: outputA}
+          output_file: '{workdir}/output'
+        step: {$ref: steps.yml#/stepB}
+
+
+In the above example, the subworkflow is called as a multistep-stage. The subworkflow is referenced using :code:`workflow: {$ref: 'subflow.yml'}` and the packtivities to be scheduled is in turn referenced by the subworkflow. The scattering is done over the variable :code:`input` creating a number of instances. Since the scope was defined as a subworkflow the instances run parallel to each other. In the example, when instance 5 is done with step one (stageA) it can continue to step two (stageB) without having to wait for instance 4 or any other instance to finish step one.
```

Comment:
```suggestion
In the above example, the subworkflow is called as a multistep-stage. The subworkflow is referenced using :code:`workflow: {$ref: 'subflow.yml'}` and the packtivities to be scheduled are in turn referenced by the subworkflow. The scattering is done over the variable :code:`input`, which creates a number of instances. Since the scope was defined as a subworkflow, the instances run in parallel. In the example, when instance 5 is done with step one (stageA) it can continue to step two (stageB) without having to wait for instance 4 or any other instance to finish step one.
```